#!/usr/bin/env python
# DWA Planner configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t
gen = ParameterGenerator()

gen.add("min_linear_velocity", double_t, 0, "The minimum desired x velocity for the robot in m/s", 0.5, 0, 1.5)
gen.add("min_lookahead_distance", double_t, 0, "The lookahead distance for path heading comparisons", 1.0, 0, 4.0)
gen.add("max_lookahead_distance", double_t, 0, "The lookahead distance for path heading comparisons", 2.0, 0, 4.0)
gen.add("max_heading_difference", double_t, 0, "The heading difference for speed limiting", 1.0, 0, 3.14)
gen.add("min_heading_difference", double_t, 0, "The minimum heading difference for speed limiting", 0.2, 0, 3.14)

exit(gen.generate("base_local_planner", "base_local_planner", "PathSpeedLimiter"))
